src/de/mod.rs

```diff
@@ -241,7 +249,38 @@ impl<'de> Deserializer<'de> {
     where
         V: Visitor<'de>,
     {
-        if self.newtype_variant || self.parser.consume_char('(') {
+        if self.extensions().contains(Extensions::BRACED_STRUCTS) {
```

Comment:
This separate if statement should probably refactored to duplicate less code with the else-if branch